OSDOCS 12344: Updated issuer types for cert-manager (#87262)
openshift/openshift-docs: modules/cert-manager-issuer-types.adoc

:_mod-docs-content-type: CONCEPT
[id="cert-manager-issuer-types_{context}"]
= {cert-manager-operator} Issuer Providers

The cert-manager operator for Red Hat OpenShift has been tested with the following issuer types:

* Automated Certificate Management Environment (ACME)
* Self-signed

> * link:https://cloud.google.com/security/products/certificate-authority-service[Google cloud Certificate Authority Service] (Google CAS)

Reviewer: Please move `Google CAS` to the botton. Becasue the NCM and Google CAS are external issuers. We doesn't and won't fully test external issuers.

It'd be better to list them (external issuers) after in-tree issuer types (ACME, CA, Self-signed, Vault, Venafi).

* Certificate authority (CA)
* Self-signed
* link:https://cloud.google.com/security/products/certificate-authority-service[Google cloud Certificate Authority Service] (Google CAS)
* link:https://cert-manager.io/docs/configuration/vault/[Vault]

> * link:https://cert-manager.io/docs/configuration/venafi/[Venafi]

Author: Some of the changes were suggested by Nick. Since venafi was acquired by cyberark. Can you check again if venafi is supported?

> * link:https://cert-manager.io/docs/configuration/venafi/[Venafi]

Reviewer: Thanks for providing the context. I'll check with Nick when he is back. Then we could adjust the wording later.

The following table outlines the test coverage for each tested issuer type:

[cols="^30,^20,^50",options="header"]
|===
| Issuer Type                          | Test Status        | Notes

| ACME                                 | Fully Tested       | Verified with standard ACME implementations.
| Self-signed                          | Fully Tested       | Ensures basic self-signed functionality.
| Google CAS          | Fully Tested       | Compatible with common CA configurations.
| link:https://cert-manager.io/docs/configuration/vault/[Vault] | Fully Tested   | Limited to standard Vault setups due to infrastructure access constraints.
| link:https://www.nokia.com/networks/security-portfolio/netguard/certificate-manager/[Nokia NetGuard Certificate Manager (NCM)] | Partially Tested   | Subject to provider-specific limitations.

> |===

Reviewer: Please also update this table to reflect the order I mentioned above.

- ACME, Fully Tested
- CA, Fully Tested
- Self-signed, Fully Tested
- Vault, Fully Tested
- Venafi, Partially Tested
- NCM, Partially Tested
- Google CAS, Partially Tested

Reviewer: Small suggestion : The link references to Issuer Type in this table can be removed safely, as they are already included in the bullet points section.

Author: ACK

:_mod-docs-content-type: CONCEPT
[id="cert-manager-issuer-types_{context}"]

> = {cert-manager-operator} Issuer Providers

Reviewer 2: ```suggestion
= {cert-manager-operator} issuer providers
```

> The cert-manager operator for Red Hat OpenShift has been tested with the following issuer types:

Reviewer 2: ```suggestion
The cert-manager operator for {product-title} has been tested with the following issuer types:
```

Reviewer: +1 here. How about using `{cert-manager-operator}` directly? (Then it will be translated into `cert-manager Operator for Red Hat OpenShift` for us)
```suggestion
The {cert-manager-operator} has been tested with the following issuer types:
```

:_mod-docs-content-type: CONCEPT
[id="cert-manager-issuer-types_{context}"]
= {cert-manager-operator} Issuer Providers

The cert-manager operator for Red Hat OpenShift has been tested with the following issuer types:

* Automated Certificate Management Environment (ACME)
* Certificate Authority (CA)
* Self-signed
* link:https://cert-manager.io/docs/configuration/vault/[Vault]
* link:https://cert-manager.io/docs/configuration/venafi/[Venafi]
* link:https://www.nokia.com/networks/security-portfolio/netguard/certificate-manager/[Nokia NetGuard Certificate Manager] (NCM)
* link:https://cloud.google.com/security/products/certificate-authority-service[Google cloud Certificate Authority Service] (Google CAS)

[id="cert-manager-issuer-types-testing_{context}"]
== Testing issuer types

The following table outlines the test coverage for each tested issuer type:

[cols="^30,^20,^50",options="header"]
|===
| Issuer Type                          | Test Status        | Notes

| ACME                                 | Fully Tested       | Verified with standard ACME implementations.

> | CA                                 | Fully Tested       | 
> | Self-signed                          | Fully Tested       | Ensures basic self-signed functionality.
> | Vault | Fully Tested   | Limited to standard Vault setups due to infrastructure access constraints.
> | Venafi          | Partially Tested       | 

Reviewer: It would be great to fill the blanks:
```suggestion
| CA                                 | Fully Tested       | Ensures basic CA functionality.
| Self-signed                          | Fully Tested       | Ensures basic self-signed functionality.
| Vault | Fully Tested   | Limited to standard Vault setups due to infrastructure access constraints.
| Venafi          | Partially Tested       | Subject to provider-specific limitations.
```

Author: As per the confirmation through this [https://redhat.enterprise.slack.com/archives/C045YMPKR3M/p1737471555720319?thread_ts=1737040886.756089&channel=C045YMPKR3M&message_ts=1737471555.720319](Slack text), I will be removing venafi from the list.
